azure-pipelines.yml

```diff
@@ -12,6 +13,7 @@ stages:
     - template: templates/commonbuildtasks.yml
     - job: BuildJob
       displayName: NuGet Build
+      dependsOn: CommonBuildJob
```

Comment:
This doesn't look like it will fix the issue. 

The version environment variable won't be defined in the pack step because it's being defined in another job (completely isolated from this job). I would move the pack task into the build job and publish that as the build stage artifact.  That will also fix the fact that it can't find the .dll (because the build is happening in another job). Then you can add `checkout: none` to this stage and simply download and push the NuGet package.